src/plugin.rs

```diff
@@ -242,20 +372,47 @@ impl<R: Runtime> Plugin<R> for TauriSql<R> {
             } else {
                 serde_json::from_value(config)?
             };
+
+            let locations = Locations::new(app);
+            println!(
+                "Locations:\n\tapp: {}\n\tresources: {}",
+                &locations.app.to_str().unwrap_or(""),
+                locations.resources.to_str().unwrap_or("")
+            );
+
             let instances = DbInstances::default();
             let mut lock = instances.0.lock().await;
             for db_url in config.preload {
-                if !Db::database_exists(&db_url).await.unwrap_or(false) {
-                    Db::create_database(&db_url).await?;
+                let conn = Connection::from(db_url)
+                    .add_app_directory(&locations.app)
+                    .to_string();
+
+                if !Db::database_exists(&conn).await.unwrap_or(false) {
+                    Db::create_database(&conn).await?;
                 }
-                let pool = Pool::connect(&db_url).await?;
-                if let Some(migrations) = self.migrations.as_mut().unwrap().remove(&db_url) {
+                let pool = Pool::connect(&conn).await?;
+
+                if let Some(migrations) = self.migrations.as_mut().unwrap().remove(&conn) {
+                    // TODO: iterate over migrations and replace the path to include the app_dir when using Sqlite
```

Comment:
This is the area in migrations that needs to mutated. You'll see some commented out code where I was flailing around trying to map over the migrations and adjust the file path in similar fashion to how it's done elsewhere but I couldn't get my groove on at all.

It might be quite simple -- in fact I was expecting it to be -- but I think I don't fully understand the data type. Alternatively, rather then _amend_ the file path here we could alternatively just have the initial configuration the consumer sends in be stored as a vector and then we just set it to the correct path here. That was a bigger change and I felt at least Fabian was more comfortable with a smaller change.

Reply:
Ok, i've put in a solution to this now. Still haven't fully validated that it's working as expected yet.